src/content/add-to-app/ios/add-flutter-screen.md

```diff
@@ -147,45 +148,39 @@ We then pass this `FlutterEngine` into a
 {% tabs "darwin-framework" %}
 {% tab "SwiftUI" %}
 
-The following example shows a generic `ContentView` with a
-`Button` hooked to present a [`FlutterViewController`][].
-The `FlutterViewController` constructor takes the pre-warmed 
-`FlutterEngine` as an argument. `FlutterEngine` is passed in 
-as an `EnvironmentObject` via `flutterDependencies`.
+The following example shows a generic `ContentView` with a 
+[`NavigationLink`][] hooked to a flutter screen. 
+First, create a `FlutterViewControllerRepresentable` to represent the 
+`FlutterViewController`. The `FlutterViewController` constructor takes 
+the pre-warmed `FlutterEngine` as an argument, which is injected through
+the view environment. 
 
 ```swift title="ContentView.swift"
 import SwiftUI
 import Flutter
 
-struct ContentView: View {
-  // Flutter dependencies are passed in an EnvironmentObject.
-  @EnvironmentObject var flutterDependencies: FlutterDependencies
-
-  // Button is created to call the showFlutter function when pressed.
-  var body: some View {
-    Button("Show Flutter!") {
-      showFlutter()
-    }
-  }
-
-func showFlutter() {
-    // Get the RootViewController.
-    guard
-      let windowScene = UIApplication.shared.connectedScenes
-        .first(where: { $0.activationState == .foregroundActive && $0 is UIWindowScene }) as? UIWindowScene,
-      let window = windowScene.windows.first(where: \.isKeyWindow),
-      let rootViewController = window.rootViewController
-    else { return }
-
-    // Create the FlutterViewController.
-    let flutterViewController = FlutterViewController(
+struct FlutterViewControllerRepresentable: UIViewControllerRepresentable {
+  // Flutter dependencies are passed in through the view environment.
+  @Environment(FlutterDependencies.self) var flutterDependencies
+  
+  func makeUIViewController(context: Context) -> some UIViewController {
+    return FlutterViewController(
       engine: flutterDependencies.flutterEngine,
       nibName: nil,
       bundle: nil)
-    flutterViewController.modalPresentationStyle = .overCurrentContext
-    flutterViewController.isViewOpaque = false
+  }
+  
+  func updateUIViewController(_ uiViewController: UIViewControllerType, context: Context) {}
+}
+
+struct ContentView: View {
 
-    rootViewController.present(flutterViewController, animated: true)
+  var body: some View {
+    NavigationStack {
+      NavigationLink("My Flutter Feature") {
+        FlutterViewControllerRepresentable()
+      }
+    }
```

Comment:
At the end of this section is this note:

![image](https://github.com/user-attachments/assets/e5049d47-c7e9-4cb5-8fea-3403f9b87aa0)

This note is incorrect for the `SwiftUI` tab (the `FlutterEngine` `run` method is called when the `FlutterDependencies` observable is initialized). Consider moving this note into the `UIKit-Swift` and `UIKit-ObjC` tabs.